### PR TITLE
Add guard against nil/error/panic for topologies.go

### PR DESCRIPTION
**Pull request by Zheaoli**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

https://github.com/kubernetes/kubernetes/issues/107415
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by Zheaoli** [comment]
This is the backport version for PR #107531

**Comment by Zheaoli** [comment]
cc @liggitt @ahg-g

File: pkg/scheduler/util/topologies.go [hunk 1]
```diff
@@ -38,6 +38,9 @@ func GetNamespacesFromPodAffinityTerm(pod *v1.Pod, podAffinityTerm *v1.PodAffini
 // PodMatchesTermsNamespaceAndSelector returns true if the given <pod>
 // matches the namespace and selector defined by <affinityPod>`s <term>.
 func PodMatchesTermsNamespaceAndSelector(pod *v1.Pod, namespaces sets.String, selector labels.Selector) bool {
+	if selector == nil {
```
Comment (liggitt) [inline pkg/scheduler/util/topologies.go:41]: this check is ok, but also add a `continue` here so we don't call into this function knowingly with a bad selector:

https://github.com/kubernetes/kubernetes/blob/aaef1f46ef84a2d7bd71364165655e05219f2606/pkg/scheduler/internal/queue/scheduling_queue.go#L545-L548

Comment (Zheaoli) [reply pkg/scheduler/util/topologies.go:41]: Sounds good! I'll update my pr

Comment (Zheaoli) [reply pkg/scheduler/util/topologies.go:41]: I have update my PR, PTAL

**Pull request closed without merging**